coverage_utils.py

```diff
@@ -0,0 +1,68 @@
+from nose.tools import set_trace
+from core.coverage import (
+    BibliographicCoverageProvider,
+    CoverageFailure
+)
+from core.model import DataSource
+
+class MetadataWranglerBibliographicCoverageProvider(BibliographicCoverageProvider):
+
+    def work(self, identifier):
+        # There should already be a dummy LicensePool, created by
+        # IdentifierResolutionCoverageProvider, which we can use as a
+        # basis for a work.
+        licensepools = identifier.licensed_through
+        if licensepools:
+            license_pool = licensepools[0]
+        else:
+            # Even if not, we can create our own LicensePool -- it's just
+            # a stand-in and doesn't represent any actual licenses.
+            license_pool = self.license_pool(
+                identifier, data_source=DataSource.INTERNAL_PROCESSING
+            )
+            if not license_pool.licenses_owned:
+                license_pool.update_availability(1, 1, 0, 0)
+
+        return super(
+            MetadataWranglerBibliographicCoverageProvider, self).work(
+                identifier, license_pool
+            )
+
+
+class ResolveVIAFOnSuccessCoverageProvider(MetadataWranglerBibliographicCoverageProvider):
+    """A mix-in class for metadata wrangler BibliographicCoverageProviders
+    that add author information. When such a coverage provider
+    completes its work, it should run any Contributors associated with
+    the presentation Edition through VIAF. Then it should try to
+    create a presentation-ready work.
```

Comment:
Why not make VIAF a separate coverage provider? Other coverage providers could tell it to run immediately or not, and then it could keep a separate record of work that was done. I guess it should be a ContributorCoverageProvider rather than an IdentifierCoverageProvider, which we probably don't have yet.

Reply:
This is a good idea.

The whole VIAF thing is another issue that turned out to be kind of marginal. Like OCLC Lookup, we overestimated its importance because we started out dealing with books which had nearly no metadata. The books we deal with now don't have as many problems. So I'm not too worried about how well VIAF lookup works until we have a way of indicating that two Contributor objects refer to the same person.